diff --git a/homeassistant/components/http/__init__.py b/homeassistant/components/http/__init__.py
index de864a0..6a41c27 100644
--- a/homeassistant/components/http/__init__.py
+++ b/homeassistant/components/http/__init__.py
@@ -44,6 +44,7 @@ CONF_SERVER_PORT = 'server_port'
 CONF_DEVELOPMENT = 'development'
 CONF_SSL_CERTIFICATE = 'ssl_certificate'
 CONF_SSL_KEY = 'ssl_key'
+CONF_SSL_PEER_AUTHORITY = 'ssl_peer_authority'
 CONF_CORS_ORIGINS = 'cors_allowed_origins'
 CONF_USE_X_FORWARDED_FOR = 'use_x_forwarded_for'
 CONF_TRUSTED_NETWORKS = 'trusted_networks'
@@ -87,6 +88,7 @@ HTTP_SCHEMA = vol.Schema({
     vol.Optional(CONF_DEVELOPMENT, default=DEFAULT_DEVELOPMENT): cv.string,
     vol.Optional(CONF_SSL_CERTIFICATE, default=None): cv.isfile,
     vol.Optional(CONF_SSL_KEY, default=None): cv.isfile,
+    vol.Optional(CONF_SSL_PEER_AUTHORITY, default=None): cv.isfile,
     vol.Optional(CONF_CORS_ORIGINS, default=[]): vol.All(cv.ensure_list,
                                                          [cv.string]),
     vol.Optional(CONF_USE_X_FORWARDED_FOR, default=False): cv.boolean,
@@ -116,6 +118,7 @@ def async_setup(hass, config):
     development = conf[CONF_DEVELOPMENT] == '1'
     ssl_certificate = conf[CONF_SSL_CERTIFICATE]
     ssl_key = conf[CONF_SSL_KEY]
+    ssl_peer_authority = conf[CONF_SSL_PEER_AUTHORITY]
     cors_origins = conf[CONF_CORS_ORIGINS]
     use_x_forwarded_for = conf[CONF_USE_X_FORWARDED_FOR]
     trusted_networks = conf[CONF_TRUSTED_NETWORKS]
@@ -134,6 +137,7 @@ def async_setup(hass, config):
         api_password=api_password,
         ssl_certificate=ssl_certificate,
         ssl_key=ssl_key,
+        ssl_peer_authority=ssl_peer_authority,
         cors_origins=cors_origins,
         use_x_forwarded_for=use_x_forwarded_for,
         trusted_networks=trusted_networks,
@@ -167,8 +171,8 @@ class HomeAssistantWSGI(object):
     """WSGI server for Home Assistant."""
 
     def __init__(self, hass, development, api_password, ssl_certificate,
-                 ssl_key, server_host, server_port, cors_origins,
-                 use_x_forwarded_for, trusted_networks,
+                 ssl_key, ssl_peer_authority, server_host, server_port,
+                 cors_origins, use_x_forwarded_for, trusted_networks,
                  login_threshold, is_ban_enabled):
         """Initialize the WSGI Home Assistant server."""
         import aiohttp_cors
@@ -191,6 +195,7 @@ class HomeAssistantWSGI(object):
         self.api_password = api_password
         self.ssl_certificate = ssl_certificate
         self.ssl_key = ssl_key
+        self.ssl_peer_authority = ssl_peer_authority
         self.server_host = server_host
         self.server_port = server_port
         self._handler = None
@@ -292,6 +297,9 @@ class HomeAssistantWSGI(object):
             context.options |= SSL_OPTS
             context.set_ciphers(CIPHERS)
             context.load_cert_chain(self.ssl_certificate, self.ssl_key)
+            if self.ssl_peer_authority:
+                context.verify_mode |= ssl.CERT_REQUIRED
+                context.load_verify_locations(self.ssl_peer_authority)
         else:
             context = None
 
